r/vignettes/install.Rmd

```diff
@@ -379,63 +312,86 @@ setting `ARROW_WITH_ZSTD=OFF` to build without `zstd`; or (3) uninstalling
-  dependencies from source.
-* `LIBARROW_BUILD`: If set to `false`, the build script
+# Summary of build environment variables
+
+## libarrow configuration
+
+Some features are optional when you build Arrow from source - you can configure
+whether these components are built via the use of environment variables.  The 
+names of the environment variables which control these features and their 
+default values are shown below.
+
+| Name  | Description   | Default Value |
+| ---| --- | :-: |
+| `ARROW_S3`          | S3 support (if dependencies are met)*   | `OFF` |
+| `ARROW_JEMALLOC`    | The `jemalloc` memory allocator        | `ON` |
+| `ARROW_MIMALLOC`    | The `mimalloc` memory allocator        | `ON` |
+| `ARROW_PARQUET`    |   | `ON` |
+| `ARROW_DATASET`    |  | `ON` |
+| `ARROW_JSON`    | The JSON parsing library | `ON` |
+| `ARROW_WITH_RE2`    | The RE2 regular expression library, used in some string compute functions | `ON` |
+| `ARROW_WITH_UTF8PROC` | The UTF8Proc string library, used in many other string compute functions  | `ON` |
+| `ARROW_WITH_BROTLI`  | Compression algorithm  | `ON` |
+| `ARROW_WITH_BZ2`    | Compression algorithm  | `ON` |
+| `ARROW_WITH_LZ4`    | Compression algorithm  | `ON` |
+| `ARROW_WITH_SNAPPY`    | Compression algorithm  | `ON` |
+| `ARROW_WITH_ZLIB`    | Compression algorithm | `ON` |
+| `ARROW_WITH_ZSTD`    | Compression algorithm | `ON` |
+
+## R package configuration
+
+There are a number of other variables that affect the `configure` script and 
+the bundled build script.  All boolean variables are case-insensitive.
+
+| Name | Description | Default |
+| --- | --- | :-: |
+| `ARROW_USE_PKG_CONFIG` | Use `pkg-config` to search for `libarrow` install | `true` |
+| `LIBARROW_BUILD` | Allow building from source | `true` |
+| `LIBARROW_BINARY` | Try to install `libarrow` binary instead of building from source | `true` |
+| `LIBARROW_MINIMAL` | Build with minimal features enabled | (unset) |
+| `NOT_CRAN` | Set `LIBARROW_BINARY=true` and `LIBARROW_MINIMAL=false` | `false` |
+| `ARROW_R_DEV` | More verbose messaging and regenerates some code | `false` |
+| `LIBARROW_DEBUG_DIR` | Directory to save source build logs | (unset) |
+| `CMAKE` | Alternative CMake path | (unset) |
+
+See below for more in-depth explanations of these environment variables.
+
+* `ARROW_USE_PKG_CONFIG`: If set to `false`, the configure script won't look for 
+Arrow libraries on your system and instead will look to download/build them.
+  Use this if you have a version mismatch between installed system libraries and
+  the version of the R package you're installing.
+* `LIBARROW_BINARY` : If set to `true`, the script will try to download a binary
+  C++ library built for your operating system.  You may also set it to some other string,  a related "distro-version" that has binaries built that work for your OS.
+  If no binary is found, installation will fall back to building C++  dependencies from source. 
```

Comment:
Could/should we link to the csv that shows what those strings are? Or give an example?

Reply:
Added a link